### ISSUE TITLE
http_cache not working

### ISSUE THREAD

**Issue by Metis77** [contao/core-bundle#1246]
after updating to 4.4.9 var/cache/prod/http_cache stopped working.
with 4.4.6 it still worked.

"Systemwartung" say 0 cached files and the folder http_cache stays empty.

tested in multiple websites, but same server.

**Comment by bytehead** [comment]
I've encountered the same.

**Comment by Toflar** [comment]
Can you test with 4.4.8 or does "with 4.4.6 it still worked" mean, that's the last one it worked with?

**Comment by Metis77** [comment]
just tested:
4.4.9 - not working
4.4.8 - not working
4.4.7 - I dont know
4.4.6 - worked

**Comment by Toflar** [comment]
Ok, I'll let you find out if 4.4.7 works then.

**Comment by Metis77** [comment]
well, updating from a Contao 4.4.4 version to 4.4.7 specifically failed using contao manager oO.
maybe another issue.

Without beeing able to install specific version with contao manager, I cant test right now.

**Comment by Toflar** [comment]
Okay after an hour of debugging, I found the issue. It was introduced by @aschempp in https://github.com/contao/core-bundle/commit/cae98ce3ec038982f47167264ab1c3464ee3b282#diff-6c590ef7e538a1f4a9115824540acb51. 
I think merging the `Cache-Control` header needs special handling. 

In my case, the `$response` correctly contained the `Cache-Control` header `public, s-maxage=10800`. In `$this->headers` the `Cache-Control` header contains `Cache-Control: max-age=0, private, must-revalidate`. The end result is `max-age=0, must-revalidate, private, public, s-maxage=10800` which is considered to be `private` and thus nothing is cached.
I have no idea why the `Cache-Control` header is even part of `headers_list()`. I debugged using `headers_list()` with the two arguments to debug who set the header but the result was an empty string and `0` which tells me it's coming natively from the server. 

There's certainly some kind of bug in Symfony because when merging the `Cache-Control` headers it shouldn't be `private` and `public` at the same time. But also I wonder if it ever makes sense to merge a cache header?

**Comment by Toflar** [comment]
It's wrong that `public` is still there but basically it does the right thing. It merges the `private` instruction from `headers_list()` onto the response which turns it `private`. I just have no clue why there's a default `Cache-Control` header present 😄

**Comment by Toflar** [comment]
It's not by default, it's set during the preflight request. For whatever reason. Nobody uses `header()` there? Man, this is fucked up as fuck. I need some sleep otherwise my brain is going to explode.

**Comment by bytehead** [comment]
Thanks for investigating. This covers my observations and I ended up quite confused 😄

**Comment by aschempp** [comment]
Doesn‘t our Template::getResponse method set everything to private if it‘s not public?

**Comment by leofeyer** [comment]
I cannot reproduce the issue on contao.org:

```sh
bin/console cache:clear

find var/cache/prod/http_cache -type f | wc -l
0
```

One minute later:

```sh
find var/cache/prod/http_cache -type f | wc -l
20
```

**Comment by Toflar** [comment]
> Doesn‘t our Template::getResponse method set everything to private if it‘s not public?

As I said, it happens during the process of merging http headers. @leofeyer what stack do you use on contao.org?

**Comment by leofeyer** [comment]
Ubuntu, Nginx and PHP-FPM with PHP 7.1.

**Comment by Metis77** [comment]
> Ubuntu, Nginx and PHP-FPM with PHP 7.1.
same stack here, but with that error.

**Comment by leofeyer** [comment]
I'm not using the managed edition on contao.org. Maybe that makes the difference?

**Comment by bytehead** [comment]
I don't use the managed edition either, but still get the wrong behaviour.

**Comment by Toflar** [comment]
I think I‘ve found the issue. I‘d like to investigate a bit more on the reasons before I work on a solution. Will report back.

**Comment by Toflar** [comment]
Can you guys check your `phpinfo()` and tell me the `session.cache_limiter` setting. I suppose it contains `no-cache` for all of you except for @leofeyer?

**Comment by bytehead** [comment]
```

**Comment by leofeyer** [comment]
Probably related: https://github.com/symfony/symfony-docs/issues/8784

**Comment by Toflar** [comment]
Okay, so here's the summary of a day of debugging and investigation:

Why does the issue suddenly appear?
------

Normally, when you call `session_start()` to start the session, it directly sets additional headers to make sure the response is `private`. This makes a lot of sense because - in theory - whenever you start a session, the response is private to you **by defintion**.
Prior to Symfony 3.4 and 4.0, setting these headers automatically was suppressed. This was considered to be a bug and fixed in the latest versions. For the record: I support this decision, this is a bug and even if it changed the behavior, it is a correct bugfix.

What effect does it have on Contao?
-----

As of now, everytime you work with the session. A response turns `private` and thus, the reverse proxy (or call it "html cache" if you like) never stores any response.
The problem we have is that we **always** work with the session. E.g. we store the `REQUEST_TOKEN` in the session, locale stuff, `$_SESSION['FORM_DATA']` when submitting a form etc. pp. The list goes on forever.

So what now?
-----

The correct solution would be to never work with the session if you don't need it. If nobody accesses the session of Symfony, the session is never started and thus that issue would not occur.
We improved that already in Contao 4.5 by having a separate cookie for the CSRF protection (Request Token) so it's not dependent on the session anymore. We would need further adjustments (`LocaleListener` for example) to drop usage of the session and we **could** eventually manage to make sure, the **Contao Core** (I don't even want to think about extensions) does not need any session in the front end anymore. However, what would that mean? Yes, exactly: this would cause that the **session is never started**. We cannot do this because that would be a huge BC break.  Everyone accessing `$_SESSION` would always get `null` because nobody started the session. In fact, nobody noticed this but `$_SESSION` being present was a **lucky** side effect of having the `RequestToken` class accessing - and thus starting - the session for every request :D 

So you see, it all comes down to this: In Contao, a session never implied a `private` response. This was wrong ever since. This is also why we keep telling users not to enable cache on certain pages (e.g. when there's a form on it). We have the same issue here again.
In Contao it was always the developers that controlled if a response is cacheable or not (by doing funny stuff like `$objPage->cache = 0;` or whatever).

So can we just restore the old behavior?
---

No. We should not adjust a Symfony-wide configuration. We could adjust the `MergeHttpHeadersListener` to just ignore `Cache-Control` headers. That would basically restore that behaviour for all Contao related routes but not affect other routes.

What would be the right solution?
---

Stop using `$_SESSION`! Stop using it, use the Symfony session (`$request->getSession()`) instead and always check if it was already started **if you don't explicitly need it**. And also make sure you understand how proxies work and if you really need a session. Do not just blindly store things in the session because - again - that kind of makes every response private and thus these responses can never be cached.
I think we could just adjust the `MergeHttpHeadersListener` but it kind of hides the real issues. We can just continue on and even when using the proper Symfony session object, nobody notices any issues with the cache because it's hidden by the `MergeHttpHeadersListener`. Remember: making all responses `private` when starting a session **is done on purpose**. There's a reason why you should not do otherwise. We just cannot quickly fix it because of BC. But maintaining BC does not mean we should not force our developers to use their brains and do things correctly 😄  In my opinion, we should maybe adjust the `MergeHttpHeadersListener` as a quick fix but I would prefer reverting that change later on, cleaning up everything in the Contao core (no more `$_SESSION`) and providing a BC layer for `$_SESSION`. We could set it to an `ArrayAccess` object that could start the session as soon as accessed. Similar to what we have already (but it would need to start the session in addition to just behaving like a bag):

```
$_SESSION['BE_DATA'] = $this->session->getBag('contao_backend');
$_SESSION['FE_DATA'] = $this->session->getBag('contao_frontend');
```

Not sure if this would work out but maybe worth a try. I just don't feel very comfortable just adjusting the listener and ignoring further issues.

**Comment by Metis77** [comment]
>Can you guys check your phpinfo() and tell me the session.cache_limiter setting. I suppose it contains nocache for all of you except for @leofeyer?

also: session.cache_limiter => nocache

**Comment by Toflar** [comment]
So providing a BC layer for `$_SESSION` does not work, I've tested that. The framework has to start the session for BC all the time 😢

**Comment by leofeyer** [comment]
This will only change in Contao 4.5.

**Comment by Toflar** [comment]
It's a bugfix that we can merge into 4.4, it doesn't change anything.

**Comment by Toflar** [comment]
https://github.com/contao/core-bundle/pull/1260

Did not adjust tests yet but can you check if that fixes the issue?

**Comment by Metis77** [comment]
@Toflar 
not sure if this will help you:
after replacing those two files from #1260 in my Contao 4.4.9 installation, http_cache stays empty.
In Contao Manager I get following error on rebuild cache:
```
In ContaoFramework.php line 303:                                   
  Call to a member function getSession() on null  
```

**Comment by Toflar** [comment]
I'll update the PR there accordingly. There's a quickfix for Contao 4.4 (https://github.com/contao/core-bundle/pull/1263) but also I think the behaviour in Symfony is inconsistent and needs to be fixed. So there's todo's on many ends in my opinion.
Ref: https://github.com/symfony/symfony/issues/24988

**Comment by leofeyer** [comment]
Fixed in beab312d9d3e9d22af45a4d7611ba738ee3dc926. It seems that the issue will be fixed in Symfony as well: https://github.com/symfony/symfony/pull/25583

**Comment by fritzmg** [comment]
This problem still exists in Contao `4.5.1` as well as Contao `4.4.12`. In my tests only the index page and one ajax request will get cached, no other pages.

**Comment by Metis77** [comment]
I can confirm, that this stopped working in Contao 4.4.12. 
Did not test 4.5.1.

**Comment by Toflar** [comment]
See my comment in the referenced ticket.

**Comment by Metis77** [comment]
shouldn't this ticket left open and been updated, until issue is solved?

I got pretty confused seeing this being closed.
What is the status? Was there already a fix / change in Symfony?
At least the update to contao 4.4.13 did not update Symfony for me.

**Comment by fritzmg** [comment]
I don't quite understand the state of this issue right now. @Toflar 's PR for Symfony has been merged and released and there was supposed to be a quickfix for Contao 4.4. However, the cache is still not working in Contao 4.4.13 for instance.

@Toflar could you clarify?

**Comment by Toflar** [comment]
The issue is that Symfony now turns responses into `private` responses as soon as the session is started. This was considered a bugfix because in regular applications, as soon as a session was started, it means that the current visitor has private content that should never be cached by a reverse proxy. Otherwise, you risk showing private content to somebody else. From the Symfony perspecitve, this was a correct decision in my opinion.
However, that doesn't mean a response cannot be shared if the session is started. It just requires a lot more work than just setting the correct cache headers. You need to be very careful, be able to purge cached entries, set correct `Vary` headers, maybe need our replay-headers-bundle etc. pp. All of what Symfony does not want to provide by default (understandably) because Symfony focusses on providing a rock-solid abstraction for request -> response, not a fully fledged http caching framework.

My PR was not supposed to be a quickfix. There's only one fix to this problem: Have a proper listener that is triggered **after** the one of Symfony which turns the `private` response into a `public` response again, if all preconditions are met. And this is a pretty challenging task and requires a lot of knowledge on how the Symfony fullstack framework application flow works. We have many complicated use cases such as the front end preview, uncacheable insert tags, mobile page layouts and probably other stuff. It's not something we can fix quickly and I guess it needs extensive discussions and proper testing before we can even think about fixing this.
So in short: Prepare for your `http_cache` not to work for quite a while until we found the right solution (which I'm convinced we will).

**Comment by Toflar** [comment]
> So in short: Prepare for your http_cache not to work for quite a while until we found the right solution (which I'm convinced we will).

BTW: You're very welcome to help working on the solution, this was rather targeted towards the Contao users, not the developers 😄

**Comment by leofeyer** [comment]
I still cannot reproduce this issue neither on contao.org nor in my local test environment.

**Comment by fritzmg** [comment]
@leofeyer: in my tests only the start page of one language and an AJAX request got cached. Any other page did not generate a cache file.

**Comment by leofeyer** [comment]
I'm not even sure that the session is the culprit anymore. Might as well be https://github.com/symfony/symfony/issues/25902.

**Comment by joeherold** [comment]
![bildschirmfoto 2018-02-09 um 15 33 45](https://user-images.githubusercontent.com/3911545/36032604-d55abee8-0dae-11e8-8d20-8a5345c810d4.png)
I really needed the server page cache to work on active projects. If anyone else is interested, I pushed it to packagist.

Link: https://packagist.org/packages/pliigo/pagecacheenabler-bundle

Is installable via manager

**Comment by joeherold** [comment]
@leofeyer Hey Leo! Is it really clever, that in the ContaoCache Class (manager Bundle -> HttpKernel), the cache options of „allow_reload“ is set to true, what means a force reload (str + alt + r) of any client will result in rebuild a new cached site? Especially with the session topic of Sy 3.4.4 a new cache file for the same site is created. If someone is willing to „hit“ a Contao page, he/she could just set up a very small script with appropriate request headers to flood the physical hd storage in continuously creating cache files and very high memory usage in forcing to bypass the page cach. I would set it to the default Symfony options and/or make ist adjustable via a config param, so that one can deside of its caching policy. 


ContaoCache:
https://github.com/contao/manager-bundle/blob/master/src/HttpKernel/ContaoCache.php#L51

Symfony HttpCache options:
https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php#L54

**Comment by aschempp** [comment]
If I remember correctly, we discussed this and agreed that it makes sense. If flooding is the only issue, we cannot prevent that anyway. Add a query parameter and just count a number to a million and you will have a million cache entries...

**Comment by Toflar** [comment]
We're open for alternative proposals though. It's basically used for our FE preview only and I didn't find another solution to disable the cache completely back then, maybe things have changed or there are better ways.

**Comment by joeherold** [comment]
@Toflar I will try it in my quick fix with disabling in FE Preview

**Comment by joeherold** [comment]
What session/cookie values are available in fe_preview to identify as such one?

**Comment by aschempp** [comment]
There is only the PHP session that authenticates the user.

**Comment by joeherold** [comment]
@aschempp @Toflar As I can See, you are using the FOSHttpCache in ContaoCache, did you try to implement an "Cache event listener" to invalidate caches for frontend sites if user is logged in to backend?

**Comment by Toflar** [comment]
Yes. It's all playing together with our terminal42/header-replay-bundle. It's quite complicated 😄 I've added a diagramm to the bundle README (https://github.com/terminal42/header-replay-bundle) so you should understand what it tries to do. Also the readme explains the usage of `allow_reload`, why it's there etc.

**Comment by joeherold** [comment]
ah, great. I will have a look

**Comment by joeherold** [comment]
@Toflar after reading hours and hours of FOS Cache Documentation... (ufff) I came up with a rather simple solution I guess. To test it I just made an inline edit of the app.php in TL_ROOT/web

That way, I can bypass the default Http Cache Proxy:
```php
<?php

/*
 * This file is part of Contao.
 *
 * Copyright (c) 2005-2018 Leo Feyer
 *
 * @license LGPL-3.0+
 */

use Contao\ManagerBundle\ContaoManager\Plugin as ManagerBundlePlugin;
use Contao\ManagerBundle\HttpKernel\ContaoCache;
use Contao\ManagerBundle\HttpKernel\ContaoKernel;
use Symfony\Component\HttpFoundation\Request;
use Symfony\Component\HttpKernel\HttpKernelInterface;

/** @var Composer\Autoload\ClassLoader */
$loader = require __DIR__.'/../vendor/autoload.php';

ManagerBundlePlugin::autoloadModules(__DIR__.'/../system/modules');

ContaoKernel::setProjectDir(dirname(__DIR__));
$kernel = new ContaoKernel('prod', false);

/**
 * Class BypassCache
 * @description Extended Contao Cache, to bypass the CacheProxy via cookie value.
 */
class BypassCache extends ContaoCache {
    public function handle(Request $request, $type = \Symfony\Component\HttpKernel\HttpKernelInterface::MASTER_REQUEST, $catch = true)
    {
        if($request->cookies->get("BE_USER_AUTH")){
            /*
             * if setting the expect header, the parent handle function will handle the request with function pass
             * function pass: Forwards the Request to the backend without storing the Response in the cache.
             * vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php on Line 178
             *
             * That way, the Request ist handled by the Application and a unknown client may not modify/renew a cache File
             * what has an impact to every user requesting a page and then getting a new one.
             *
             * So, if a client would set the Cookie-Value of BE_USER_AUTH himself, it will only result in handling the request by
             * the application. If the value in the self created cookie value is not known, the application will remove it
             * anyway.
            */
            $request->headers->set('expect', true);
        }
        $response =  parent::handle($request, $type, $catch);
        return $response;
    }

    protected function getOptions()
    {
        return ['allow_reload' => false];
    }
}


// Enable the Symfony reverse proxy
//$kernel = new ContaoCache($kernel);
$kernel = new BypassCache($kernel);
Request::enableHttpMethodParameterOverride();

// Handle the request
$request = Request::createFromGlobals();
$response = $kernel->handle($request);
$response->send();
$kernel->terminate($request, $response);

```

**Comment by leofeyer** [comment]
@joeherold Great find! Thank you for your time.

https://github.com/symfony/symfony/blob/4ef0b3e18047a2b307af30494aa5f731e180a5f6/src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php#L178-L179

The `expect` header seems a better solution to me than `allow_reload`.

**Comment by joeherold** [comment]
I think so too. And as a side benefit: the logged in user will see the latest markup always, even if the user watches the site no only in FE Preview.

**Comment by joeherold** [comment]
@Toflar @aschempp  one more thing... sorry
would it not be a good step, to take the cache one step further and take advantage of the tagged cache? As far, as we are using the Symfony Proxy Cache, this should be possible (Varnish & Symfony Proxy) That way, cache files may be invalidated by changes e.g. of ce_elements or the article itself. I mean, that would be cool....!

https://github.com/FriendsOfSymfony/FOSHttpCacheBundle/blob/master/Resources/doc/features/tagging.rst

**Comment by Toflar** [comment]
Thanks for taking the time! It's good to discuss this with more people that have a different point of view. Anyway, I know about all of what you've mentioned, I have check it a billion times too 😄
Unfortunately, your proposal doesn't change anything at all:

The `expect` header is not different from the `allow_reload` in any way at all. Everybody can set the `expect` header from the outside so hammering the application with requests is just as simple as it is now. Basically, you just moved the problem to a different header 😄 

And yes, I'm working on cache tagging. It relies on my [toflar/psr6-symfony-http-cache-store](https://github.com/Toflar/psr6-symfony-http-cache-store) library and the integration into FOSCache and the bundle is my work too 😄

**Comment by joeherold** [comment]
@Toflar Id does change something, because it does not result into creating new cach entries, that therefor will be new to other clients... ;) So we are not touching the cache that way

**Comment by Toflar** [comment]
I see, this is an issue for the header-replay-bundle then.

**Comment by joeherold** [comment]
@Toflar so think about the following: If I am at contao.org and I will open my chrome developer tools with preserve cache in network tab by default: this will create new cache entries all the time ;)

Even hitting ctrl + shift +r is pretty common, and will create new cache entries

so i believe, there is a big difference between allow_reload and expect header

**Comment by Toflar** [comment]
Yeah, I understand. Please file an issue for the replay-header-bundle.

**Comment by joeherold** [comment]
ok, will do so

**Comment by joeherold** [comment]
but, ContaoCache is in Manager Bundle

**Comment by Toflar** [comment]
Nothing here is related to any kernel work at all. The header replay bundle takes care of all of that so that adjustments of the kernel are not needed and so that it works for real reverse proxies such as Varnish as well. We cannot just adjust the kernel class.

**Comment by joeherold** [comment]
Ok, thats above my knowledge... LOL As I understood so far, to bypass the ProxyCache, you have to set the expect header before the handler, otherwise your interaction will never reach the header bundle because of beeing served by the Proxy Cache.
But I am sure, you know better

**Comment by Toflar** [comment]
Yes, you understood correctly. What we do is we intercept and execute a "preflight" request **BEFORE** the request is sent to the kernel. So basically it does what your code does but in a more flexible, abstract way and extensible for anybody. Check the header-replay-bundle README again and check the corresponding listeners in the core bundle 😄

**Comment by Toflar** [comment]
Issue is here: https://github.com/terminal42/header-replay-bundle/issues/12
Working on it right now.

**Comment by joeherold** [comment]
hehe, so we have two issus now... LOL

https://github.com/terminal42/header-replay-bundle/issues/12

https://github.com/terminal42/header-replay-bundle/issues/11

**Comment by joeherold** [comment]
But I gave it a description... :)

**Comment by leofeyer** [comment]
The `allow_reload` option is set in the kernel class of the manager bundle, so we have to adjust this as well.

**Comment by joeherold** [comment]
hehe, that's what I was mentioning, it is related to manager bundle

**Comment by Toflar** [comment]
Yes, can you help me test 1.4.2 of the replay headers bundle together with the removal of the config?

**Comment by joeherold** [comment]
who? leo or me?

**Comment by Toflar** [comment]
Everybody can help.
What you need to do is adjust your `composer.json` to require

`"terminal42/header-replay-bundle": "dev-hotfix/1.4.2"`

and remove the `getOptions()` method in the `ContaoKernel` so that `allow_reload` is not set anymore.

**Comment by joeherold** [comment]
@leofeyer 

To solve the Cache Problem and reinvoke the Cache Creation, add a response listener after the Symfony Session listener, that sets the request to private since 3.4.4

I've done so, by adding a custom request listener with the priority of -9999, that is targeted AFTER the Symfony session listener and setting the response headers again.

![image](https://user-images.githubusercontent.com/3911545/36144804-9d909402-10af-11e8-9c58-ec60062c2c58.png)

https://bitbucket.org/joeherold/pagecacheenabler-bundle/src/1bb1fadc8ebe64c64982df2c49422d04360d04d9/src/EventListener/PageCacheEnablerListener.php?at=1.1.5&fileviewer=file-view-default

**Comment by Toflar** [comment]
Yeah we might need to do so. But also I would like to make this dependent on the `global $objPage`, so essentially moving the `FrontendTemplate::setCacheHeaders()` into that listener. Would you like to work on a Pull Request for the Core-Bundle for that? That would be awesome 😄

**Comment by joeherold** [comment]
@Toflar If you donate me a Coffee in Salzburg, sure :)

**Comment by Toflar** [comment]
That should be doable :P

**Comment by joeherold** [comment]
@Toflar 
NOT: and remove the getOptions() method in the ContaoKernel so that allow_reload is not set anymore
OK: and remove the getOptions() method in the ContaoCache so that allow_reload is not set anymore

**Comment by Toflar** [comment]
Yeah I figured you find your way around yourself :-D

**Comment by joeherold** [comment]
@Toflar hmm, My contao core-bundle is downgraded to 4.4.12 from 4.4.13 when installing `"terminal42/header-replay-bundle": "dev-hotfix/1.4.2 as 1.4.2"`

why that?

**Comment by Toflar** [comment]
I guess that's because the core-bundle 4.4.13 requires a different version. Maybe you can trick composer and also require this:

`"contao/core-bundle": "4.4.13 as 4.4.12"`

**Comment by joeherold** [comment]
@Toflar what I can tell you so far: Tested dev-hotfix/1.4.2 as 1.4.2 with contao 4.4.12, removed getOptions function in ContaoCache.php AND:
IT WORKS LIKE A CHARME

**Comment by joeherold** [comment]
can' test with 4.4.13 with the as helper...
```
- Installation request for contao/core-bundle 4.4.13 as 4.4.12 -> satisfiable by contao/core-bundle[4.4.13].
- contao/core-bundle 4.4.13 requires terminal42/header-replay-bundle 1.3.* -> satisfiable by terminal42/header-replay-bundle[1.3.0] but these conflict with your requirements or minimum-stability.
```

**Comment by Toflar** [comment]
Can you also check if the autologin functionality works for you? https://github.com/terminal42/header-replay-bundle/issues/8 is the reason why 4.4.13 requires header-replay-bundle 1.3 but it works for me with all 1.4.* versions so there's no reason to me to stick with the old version :)

**Comment by joeherold** [comment]
@Toflar works, no problem so far with Contao 4.4.13 and `"terminal42/header-replay-bundle": "dev-hotfix/1.4.2 as 1.3.2"`

**Comment by joeherold** [comment]
@Toflar freue mich auf den Kaffee... :)

**Comment by fenepedia** [comment]
@joeherold @Toflar Ich spendier euch beiden den Kaffee… Meldet euch einfach in Salzburg bei mir Christian von der Contao Academy

**Comment by Metis77** [comment]
Thanks for all the work on all the different tasks.

After updating it is still not working.
But the ticket is closed again, so it should work?

I updated with Contao Manager to Contao v4.4.14 and Symphony v3.4.4.

Did I miss anything?

```
  - Updating symfony/polyfill-mbstring (v1.6.0 => v1.7.0): Downloading (100%)
  - Updating symfony/polyfill-util (v1.6.0 => v1.7.0): Downloading (100%)
  - Updating symfony/polyfill-php70 (v1.6.0 => v1.7.0): Downloading (100%)
  - Updating symfony/polyfill-php56 (v1.6.0 => v1.7.0): Downloading (100%)
  - Updating symfony/symfony (v3.4.3 => v3.4.4): Downloading (100%)
  - Updating symfony/polyfill-intl-icu (v1.6.0 => v1.7.0): Downloading (100%)
  - Updating symfony/polyfill-apcu (v1.6.0 => v1.7.0): Downloading (100%)
  - Updating webmozart/assert (1.2.0 => 1.3.0): Downloading (100%)
  - Updating php-http/discovery (1.3.0 => 1.4.0): Downloading (100%)
  - Updating friendsofsymfony/http-cache (2.1.0 => 2.1.2): Downloading (100%)
  - Updating terminal42/header-replay-bundle (1.3.0 => 1.4.2): Downloading (100%)
  - Updating tecnickcom/tcpdf (6.2.13 => 6.2.17): Downloading (100%)
  - Updating matthiasmullie/path-converter (1.1.0 => 1.1.1): Downloading (100%)
  - Updating matthiasmullie/minify (1.3.58 => 1.3.59): Downloading (100%)
  - Updating contao-components/simplemodal (2.0.7 => 2.0.8): Downloading (100%)
  - Updating contao/core-bundle (4.4.13 => 4.4.14): Downloading (100%)
  - Updating contao/calendar-bundle (4.4.13 => 4.4.14): Downloading (100%)
  - Updating contao/faq-bundle (4.4.13 => 4.4.14): Downloading (100%)
  - Updating nelmio/security-bundle (2.4.0 => 2.5.0): Downloading (100%)
  - Updating lexik/maintenance-bundle (v2.1.3 => v2.1.5): Downloading (100%)
  - Updating contao/installation-bundle (4.4.13 => 4.4.14): Downloading (100%)
  - Updating contao/manager-bundle (4.4.13 => 4.4.14): Downloading (100%)
  - Updating contao/news-bundle (4.4.13 => 4.4.14): Downloading (100%)
```

**Comment by joeherold** [comment]
@Metis77 use this: https://packagist.org/packages/pliigo/pagecacheenabler-bundle

**Comment by Metis77** [comment]
While I was hoping for a contao built in solution, pagecacheenabler-bundle workes for me. Thank you for that!

I would still like to know, what the status of this issue is?
I know it has been closed, twice. And all the other issues I could find that are related to this are closed, too.
But still it was not finally solved, right?

Is there any other open issue I should follow, to keep updated on this? Maybe something on the Symphony side?

It is really not my intention to be inconvenient, but I feel that the cache not working is a major subject. Users should at least be able to follow current status in any open issue.

**Comment by Toflar** [comment]
The issue is still open so why should it be fixed? There are a few fixes needed, they are all mentioned here: https://github.com/contao/core-bundle/issues/1389

**Comment by Metis77** [comment]
@Toflar great thanks!
I see, there is much to come.
The tagged cache thing sounds awesome!

**Comment by leofeyer** [comment]
The issue has eventually been fixed in Contao 4.4.15 and Contao 4.5.5.

**Comment by iGweb** [comment]
Hello,

I am on a 4.5.6 version of contao.
When I configure the cache to 60 minutes on my site (https://www.igweb.fr)
it's always 15 minutes away. (you can test here: https://www.giftofspeed.com/cache-checker/)

Do you have an idea of the problem ?

thank you in advance
Samuel

**Comment by Toflar** [comment]
Are you talking about the client (your browser) or the reverse proxy cache (`var/cache/http_cache`) time?

**Comment by iGweb** [comment]
client (your browser)

**Comment by iGweb** [comment]
Ah OK ! Thank you for that answer. I will wait a bit ... ;-)

Thank you

**Comment by iGweb** [comment]
Hello,

Is there a way around the problem while waiting for the fix?

Thank you

**Comment by Toflar** [comment]
Unfortunately, there's none.

**Comment by iGweb** [comment]
Hello,
is the use of "cloudflare" a good alternative to my problem? (https://www.cloudflare.com)

Thank you

**Comment by Metis77** [comment]
@iGweb 
You should not chain caches – so if using cloudflare, you should not use contao cache at all.

Cloudflare free plan caches for 2h on client side.
Cloudflare http cache needs to be enabled by PageRules – otherwise it will only cache assets.

Cloudflare below enterprise plan does not distinguish between browser user agents, like contao does.
That is not a problem, if you do not use that - for example for the CSS classes set on body tag.

**Comment by iGweb** [comment]
ok thanks for these precision :-)

**Comment by aschempp** [comment]
> You should not chain caches – so if using cloudflare, you should not use contao cache at all.

Actually, Contao should allow you to bypass the local cache and still use the site configuration. See https://github.com/contao/manager-bundle/issues/62

**Comment by iGweb** [comment]
Hello,

I have the same problem on one of my website: https://www.jalmalv85.fr
Contao: 4.4.20

To be able to access the site in HTTPS version I have to comment the following line in the file web/app.php:
// Enable the Symfony reverse proxy
//$kernel = new ContaoCache($kernel);
Request::enableHttpMethodParameterOverride();

Is a problem?
Is there a solution to correct this?

Thank you

**Comment by aschempp** [comment]
> Is a problem?
> Is there a solution to correct this?

The only problem is that the file will be overwritten on a Contao update. But https://github.com/contao/manager-bundle/pull/71 has been merged so this feature will be available in Contao 4.6

**Comment by iGweb** [comment]
Ok Thank you
